jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java

```diff
@@ -533,11 +540,94 @@ public void abortWorkItem(WorkItem workItem,
     }
 
     private ClassLoader getInternalClassLoader() {
-        if (this.classLoader != null) {
-            return this.classLoader;
+        /* CXF builds compiler classpath assuming that the hierarchy of ClassLoader is composed of URLClassLoader instances.
+         * Since ModuleClassLoader does not implement URLClassLoader, we need to provide an alternative way of retrieving these URLS
+         * so CXF can build a proper classpath, avoiding the issue mentioned below. 
+         * @see https://issues.apache.org/jira/browse/CXF-7925
+         */
+        ClassLoader cl = this.classLoader != null ? classLoader : Thread.currentThread().getContextClassLoader(), parent = cl;
+        Collection<File> uris = new HashSet<>();
+        do {
+            if (parent.getClass().getSimpleName().equals("ModuleClassLoader")) {
+                try {
+                    getJarsFromModuleClassLoader(parent, uris);
+                } catch (ReflectiveOperationException e) {
+                    throw new IllegalStateException("Problem calculating list of URLs from ModuleClassLoader", e);
+                }
+            }
+            parent = parent.getParent();
+        } while (parent != null);
+        if (!uris.isEmpty()) {
+            cl = new CXFJavaCompileClassLoader(uris, cl);
+        }
+        return cl;
+    }
+
+    private static class CXFJavaCompileClassLoader extends URLClassLoader {
+
+        private URL[] jarUrls;
+
+        public CXFJavaCompileClassLoader(Collection<File> files, ClassLoader parent) {
+            super(new URL[0], parent);
+            // maybe it makes sense to filter only jaxb ones (there are 256 jars in kie-server)?
+            this.jarUrls = files.stream().map(CXFJavaCompileClassLoader::toUrl).toArray(URL[]::new);
```

Comment:
Yep, as per comment, maybe it's worth filtering filenames with "jaxb"

Reply:
this is a comment to easily locate the piece of code to change if, in future, we need to generate a smaller classpath for compiler.

Reply:
it should be all of them. That comment should be removed. Please try to avoid this sort of comments. If you want to point out something to reviewers just add a comment in github pull request.